keps/sig-architecture/617-improve-kep-implementation/README.md

```diff
@@ -0,0 +1,221 @@
+short.
+
+### 2) Change the names of KEPs
+
+The date-format in the name is not very human friendly and doesn't do anything
+for machines.  Instead, we propose to use the issue number of the enhancement
+issue as the KEP number.  KEP names would now be:
+`1234-short-human-friendly-string`.  This makes them easier to reference and
+remember: "KEP 1234" is easier than "KEP 20190903".  We will not retroactively
+change existing KEPs, so as to not break external links to them.
+
+### 3) Move KEPs from flat-files to a directory structure
+
+It's sometimes useful to have secondary documents alongside a KEP, for example
+image files.  To make this cleaner, new KEPs will be created in their own
+sub-directories of the SIG-specific directories. As per the above, a KEPs
+directory would be named as `1234-short-human-friendly-string`.  Inside that
+directory, the body of the KEP would be `README.md`.  Supporting documents and
+images would go in that directory.
+
+<<[UNRESOLVED]>>
+The simplest path would be to not retroactively move existing KEP files, to
+preserve inbound links.  On the other hand, we could make new dirs for existing
+KEPs, which would let us extract the metadata into `kep.yaml` which would be
+more consistent for tooling.  We could symlink the old names to the new
+`subdir/README.md` for inbound links.
+<<[/UNRESOLVED]>>
+
+### 4) Metadata split from KEP text
+
+Today, KEP metadata is a header on the markdown file.  This proposal moves that
+metadata to a new `kep.yaml` in each KEP directory (see above).  This file will
+contain the same information that was previously in the YAML front-matter.
+
+### 5) Remove some fields from metadata
+
+Almost no KEPs actually set `editor` at all.  This proposes to remove it from
+the template.
+
+The `last-updated` field can be derived from git history.
+
+The `superseded-by` field is never set and can be implied by the corresponding
+`replaces` field.
+
+### 6) Formalize one-enhancement == one-KEP
+
+There has been debate about whether a new KEP is needed to move a feature from beta
+to GA.  This proposal defines that one feature or enahancement gets one KEP,
+which is updated over its whole lifecycle.
+
+### 7) Adjust GitHub tooling for KEPs
+
+- [x] Move existing KEPs into [k/features]
+- [x] Create a `kind/kep` label for [k/community] and [k/features]
+  - For `k/community`:
+    - [ ] Label incoming KEPs as `kind/kep`
+    - [ ] Enable searches of `org:kubernetes label:kind/kep`, so we can identify active PRs to `k/community` and reroute the PR authors to `k/enhancements` (depending on the state)
+  - For `k/enhancements` (fka `k/features`):
+    - [ ] Label incoming KEPs as `kind/kep`
+    - [ ] Classify KEP submissions / tracking issues as `kind/kep`, differentiating them from `kind/feature`
+- [x] Move existing design proposals into [k/features]
+- [ ] Move existing architectural documents into [k/features] (process TBD)
+- [ ] Deprecate design proposals
```

Comment:
I'd really like to move the design proposals out of the community repo. We still have open issues and PRs and occasionally get folks proposing new ones over there =/

Reply:
+1. Original plan was to blockade the k/community design proposals directory and it probably makes sense to move on that.

Reply:
AI(justaugustus) :)